network/test_network.go

```diff
@@ -73,23 +73,12 @@ func (*noopListener) Addr() net.Addr {
 	}
 }
 
-func NewTestNetwork(
-	log logging.Logger,
+func NewTestNetworkConfig(
 	networkID uint32,
 	currentValidators validators.Manager,
 	trackedSubnets set.Set[ids.ID],
-	router router.ExternalHandler,
-) (Network, error) {
+) (*Config, error) {
 	metrics := prometheus.NewRegistry()
```

Comment:
I think this should be passed in by the caller so that the same metrics instance can be passed into `NewTestNetwork`

Reply:
Good call. Made this change.